src/datatrove/pipeline/readers/videotriplet.py

```diff
@@ -0,0 +1,170 @@
+import json
+import warnings
+import os
+from typing import Callable, List, Dict
+from datatrove.io import DataFileLike, DataFolderLike, get_datafolder, download_file
+from datatrove.pipeline.readers.base import BaseDiskReader
+from datatrove.data import Document, Media, MediaType
+from datatrove.data import Document, DocumentsPipeline
+
+class VideoTripletReader(BaseDiskReader):
+    """Read triplets of video, metadata, and optional caption files."""
+
+    name = "🎥 Video Triplet Reader"
+
+    def __init__(
+        self,
+        data_folder: DataFolderLike,
+        paths_file: DataFileLike | None = None,
+        metadata_origin: str | None = None,
+        limit: int = -1,
+        skip: int = 0,
+        file_progress: bool = False,
+        doc_progress: bool = False,
+        adapter: Callable = None,
+        text_key: str = "text",
+        id_key: str = "id",
+        default_metadata: dict = None,
+        recursive: bool = True,
+        glob_pattern: str | None = None,
+        local_cache_dir = "/tmp/local_video_cache"
+    ):
+        self.data_folder = get_datafolder(data_folder)
+        self.paths_file = paths_file
+        self.metadata_origin = metadata_origin
+        self.local_cache_dir = local_cache_dir  
+        os.makedirs(self.local_cache_dir, exist_ok=True)
+        super().__init__(
+            data_folder,
+            paths_file,
+            limit,
+            skip,
+            file_progress,
+            doc_progress,
+            adapter,
+            text_key,
+            id_key,
+            default_metadata,
+            recursive,
+            glob_pattern,
+        )
+
+
+    def run(self, data: DocumentsPipeline, rank: int = 0, world_size: int = 1) -> DocumentsPipeline:
+        """Overrides the base run method to handle triplet statistics correctly."""
+        triplet_count = 0
+
+        for triplet in self.find_triplets():
+            document = self.process_triplet(triplet)
+            if document:
+                self.stat_update("documents")  # Track the number of triplets processed
+                self.update_doc_stats(document)
+                triplet_count += 1
+                yield document
+
+    def find_triplets(self) -> List[Dict[str, str]]:
+        """Find triplets of video, metadata, and caption files in the data folder."""
+        triplets = []
+        video_extensions = (".mp4", ".avi", ".mkv", ".mov")
+        metadata_extension = ".json"
+        caption_extension = ".vtt"
+
+        if self.paths_file:
+            with self.data_folder.open(self.paths_file, "r") as f:
+                paths = [line.strip() for line in f]
+        else:
+            paths = self.data_folder.list_files(recursive=self.recursive, glob_pattern="*")
```

Comment:
if you don't want to use the glob_pattern you can just pass None (or nothing as it is the default), but then you probably want to remove it from the `__init__` args